### DISCUSSION TITLE
Bare-metal installation - confusing documentation

### DISCUSSION THREAD

**Discussion by azzid** [okd-project/okd#1910] [Q&A]
I'm attempting to install OKD in my homelab. I have three desktop (i.e. no BMC) machines (8 core/16 gb mem/500 gb disk) that I want to clusterify.

I start my journey on the[ bare-metal docs](https://docs.okd.io/latest/installing/installing_bare_metal/preparing-to-install-on-bare-metal.html) and decide on going the interactive installation route using [Assisted Installer](https://access.redhat.com/documentation/en-us/assisted_installer_for_openshift_container_platform).

Following that link however I end up on openshift documentation - no longer is there any mention of Fedora CoreOS or OKD. I followed that rabbit hole into [console.redhat.com](https://console.redhat.com) and started creating a cluster there - but that seem to give me openshift, not OKD. Discovery iso runs RHCOS and so forth.

Where can I find a better resource for doing the installation?

[This suggestion feels a bit user hostile](https://hackmd.io/tDnHM2BoQru0VgqMGI_Q5g)

**Comment by azzid** [comment]
More confusion here: https://docs.okd.io/latest/architecture/architecture-installation.html#supported-platforms-for-openshift-clusters_architecture-installation

Documentation lists `Bare metal` as one install option - but it is not in the list when running `openshift-install`:

```
➜  openshift-install create cluster
? SSH Public Key /Users/azzid/.ssh/id_rsa.pub
? Platform  [Use arrows to move, type to filter, ? for more help]
> alibabacloud
  aws
  azure
  gcp
  ibmcloud
  nutanix
  openstack
  powervs
  vsphere
```

[Seems I'm not the only one confused](https://github.com/okd-project/okd/discussions/1843)

**Comment by azzid** [comment]
The more I read the more confused I get.

When installing ocp using the tools on console.redhat.com I can stand up a cluster consisting of control-plane and worker nodes only. No load balancers needed in front of the cluster - rather the cluster itself manages the ips for `*.apps.<cluster name>.<domain>` and `api.<cluster name>.<domain>` - but when I read [the docs](https://docs.okd.io/latest/installing/installing_bare_metal/installing-restricted-networks-bare-metal.html#installation-load-balancing-user-infra_installing-restricted-networks-bare-metal) it states I must provide external load balancing when providing my own infrastructure.

Should not the api/apps virtual ips be part of the platform regardless?

The [IPI docs](https://docs.okd.io/latest/installing/installing_bare_metal_ipi/ipi-install-overview.html#ipi-install-overview) describes keepalived/haproxy for the api endpoint as running on the nodes - am I to understand that in the absence of IPI those components won't be embedded with the nodes? 

[Seems I'm not the only one confused](https://github.com/okd-project/okd/discussions/1859)

**Comment by azzid** [comment]
It is stated [here](https://docs.okd.io/latest/installing/installing_bare_metal/installing-restricted-networks-bare-metal.html#installation-initializing-manual_installing-restricted-networks-bare-metal) that:

> 2. Customize the sample install-config.yaml file template that is provided and save it in the <installation_directory>.

There is no sample `install-config.yaml` provided?

**Comment by azzid** [comment]
> If you do not already have an existing solution for a container image registry, subscribers of OKD are provided a [mirror registry for Red Hat OpenShift](https://docs.okd.io/latest/installing/disconnected_install/installing-mirroring-creating-registry.html#installing-mirroring-creating-registry). The mirror registry for Red Hat OpenShift is included with your subscription and is a small-scale container registry that can be used to mirror the required container images of OKD in disconnected installations.

[ref](https://docs.okd.io/latest/installing/disconnected_install/installing-mirroring-installation-images.html#installing-mirroring-installation-images)

There is no subscription for OKD, right? Seems this article was search-replaced?

**Reply by rbaumgar** [reply]
Red Hat OpenShift Mirror registry is part of OpenShift and not part of OKD.

**Reply by eb4x** [reply]
[mirror-registry](https://github.com/quay/mirror-registry/releases) is just a stripped-down version of quay, is relatively easy to install and works well with OKD. Just grab the offline installer where all container-images come bundled in the archive.
